db/migrate/20240304090449_migrate_interaction_settings_to_policy.rb

```diff
@@ -4,46 +4,50 @@ class MigrateInteractionSettingsToPolicy < ActiveRecord::Migration[7.1]
   disable_ddl_transaction!
 
   # Dummy classes, to make migration possible across version changes
-  class Account < ApplicationRecord
-    has_one :user, inverse_of: :account
-    has_one :notification_policy, inverse_of: :account
-  end
-
   class User < ApplicationRecord
-    belongs_to :account
+    belongs_to :notification_policy, foreign_key: 'account_id', optional: true, inverse_of: false
   end
 
-  class NotificationPolicy < ApplicationRecord
-    belongs_to :account
-  end
+  class NotificationPolicy < ApplicationRecord; end
 
   def up
-    User.includes(account: :notification_policy).find_each do |user|
-      deserialized_settings = Oj.load(user.attributes_before_type_cast['settings'])
+    User.includes(:notification_policy).in_batches do |users|
```

Comment:
If I recall from the original migration - the audience of records we want here is like "Users who have selected one of the settings UI prefs before the notification policy migrations have run", right?

If so, could we even more narrowly select only the relevant users here, or is that impossible (or too complex or edge-casey) given the settings serialization.

Reply:
Settings are serialized as a JSON blob in a `text` column, so I guess we could cast and do stuff in PostgreSQL, but I'm not sure this is worth it.